dvc-checkpoints-mnist/branches/full_pipeline/train.py

```diff
@@ -0,0 +1,121 @@
+"""Model training and evaluation."""
+import json
+import os
+
+import dvclive
+import torch
+import torch.nn.functional as F
+import torchvision
+import yaml
+
+EPOCHS = 10
+
+
+class ConvNet(torch.nn.Module):
+    """Toy convolutional neural net."""
+
+    def __init__(self):
+        super().__init__()
+        self.conv1 = torch.nn.Conv2d(1, 8, 3, padding=1)
+        self.maxpool1 = torch.nn.MaxPool2d(2)
+        self.conv2 = torch.nn.Conv2d(8, 16, 3, padding=1)
+        self.dense1 = torch.nn.Linear(16 * 14 * 14, 32)
+        self.dense2 = torch.nn.Linear(32, 10)
+
+    def forward(self, x):
+        x = F.relu(self.conv1(x))
+        x = self.maxpool1(x)
+        x = F.relu(self.conv2(x))
+        x = x.view(-1, 16 * 14 * 14)
+        x = F.relu(self.dense1(x))
+        x = self.dense2(x)
+        return x
+
+
+def transform(dataset):
+    """Get inputs and targets from dataset."""
+    x = dataset.data.reshape(len(dataset.data), 1, 28, 28) / 255
+    y = dataset.targets
+    return x, y
+
+
+def train(model, x, y, lr, weight_decay):
+    """Train a single epoch."""
+    model.train()
+    criterion = torch.nn.CrossEntropyLoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=lr, weight_decay=weight_decay)
```

Comment:
@dberenbaum isn't this missing `optimizer.load_state_dict()` to ensure the epoch number & weight decay are used correctly?